main.tf

```diff
@@ -201,7 +201,7 @@ resource "aws_cloudfront_distribution" "this" {
   }
 
   dynamic "custom_error_response" {
-    for_each = var.custom_error_response
+    for_each = [var.custom_error_response]
```

Comment:
The way I work @aokomorowski  is I feed a list of my own custom_error_response and so I never needed this change you are proposing. My example:

custom_error_response = local.error_behaviors
locals {
  error_behaviors = [
    for code, response in var.error_codes : {
      error_code            = code
      response_code         = response
      response_page_path    = "/err-page/err-img",
      error_caching_min_ttl = 1
    }
  ]
}

variable "error_codes" {
  type        = map
  description = "Error Code vs Response Codes"
  default = {
    "400" = "400",
    "403" = "403",
    "404" = "404"
  }
}